app/Controllers/Http/UsersController.ts

```diff
@@ -1,3 +1,450 @@
-// import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
+import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext';
+import { schema, rules } from '@ioc:Adonis/Core/Validator';
+import User from 'App/Models/User';
+import { isRegistrationEnabled } from '../../../config/app';
 
-export default class UsersController {}
+const newPostSchema = schema.create({
+  firstname: schema.string(),
+  lastname: schema.string(),
+  email: schema.string([
+    rules.email(),
+    // rules.unique({ table: 'users', column: 'email' }),
+  ]),
+  password: schema.string([rules.minLength(8)]),
+});
+
+export default class UsersController {
+  // Register a new user
+  public async signup({ request, response, auth }: HttpContextContract) {
+    if (isRegistrationEnabled === 'false') {
+      return response.status(401).send({
+        message: 'Registration is disabled on this server',
+        status: 401,
+      });
+    }
+
+    // Validate user input
+    let data;
+    try {
+      data = await request.validate({ schema: newPostSchema });
+    } catch (error) {
+      return response.status(401).send({
+        message: 'Invalid POST arguments',
+        messages: error.messages,
+        status: 401,
+      });
+    }
+
+    // Create user in DB
+    let user;
+    try {
+      user = await User.create({
+        email: data.email,
+        password: data.password,
+        username: data.firstname,
+        lastname: data.lastname,
+      });
+    } catch {
+      return response.status(401).send({
+        message: 'E-Mail address already in use',
```

Comment:
Couldn't this request also fail due to the password not meeting the min length of 8?

Reply:
No, in fact that would trigger the catch on line 31 😄